packages/widgetbook/lib/src/fields/field_codec.dart

```diff
@@ -32,18 +32,33 @@ class FieldCodec<T> {
     return '{${pairs.join(',')}}';
   }
 
+  /// Tring to decode [component] if possible.
+  ///
+  /// If not possible returned null.
+  static String? tryDecodeComponent(String component) {
+    try {
+      return Uri.decodeComponent(component);
+    } on ArgumentError {
+      return null;
+    }
+  }
+
   /// Decodes a query group encoded value back to a [Map].
   static Map<String, String> decodeQueryGroup(String? group) {
     if (group == null) return {};
 
     final params = group.substring(1, group.length - 1).split(',');
+    final filteredParams =
+        params.where((p) => p.split(':').length == 2).map((p) => p.split(':'));
 
     return Map<String, String>.fromEntries(
-      params.map(
-        (param) {
-          final parts = param.split(':');
-          final String decodedKey = Uri.decodeComponent(parts[0]);
-          return MapEntry(decodedKey, parts[1]);
+      filteredParams.map(
+        (parts) {
+          final decodedKey =
+              parts[0].isNotEmpty ? tryDecodeComponent(parts[0]) : parts[0];
+          final decodedValue =
+              parts[1].isNotEmpty ? tryDecodeComponent(parts[1]) : parts[1];
```

Comment:
**suggestion:** If `parts[...]` is empty, then `tryDecodeComponent` will still return an empty string, so this check is redudant.

```suggestion
          final decodedKey = tryDecodeComponent(parts[0]) ?? parts[0];
          final decodedValue = tryDecodeComponent(parts[1]) ?? parts[1];
```

Reply:
Removed check